src/main/kotlin/me/zeroeightsix/kami/feature/module/BaritoneRemote.kt

```diff
@@ -0,0 +1,58 @@
+package me.zeroeightsix.kami.feature.module
+
+import baritone.api.BaritoneAPI
+import io.github.fablabsmc.fablabs.api.fiber.v1.annotation.Setting
+import me.zero.alpine.listener.EventHandler
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.BaritoneIntegration
+import me.zeroeightsix.kami.KamiMod
+import me.zeroeightsix.kami.event.PacketEvent
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.util.Friends
+import me.zeroeightsix.kami.util.MessageDetectionHelper
+import me.zeroeightsix.kami.util.MessageDetectionHelper.detectAndRemove
+import me.zeroeightsix.kami.util.Regexes
+import me.zeroeightsix.kami.util.text
+import net.minecraft.network.packet.c2s.play.ChatMessageC2SPacket
+import net.minecraft.util.Formatting
+
+@Module.Info(
+    name = "BaritoneRemote",
+    category = Module.Category.MISC,
+    description = "Remotely control Baritone with /msg"
+)
+object BaritoneRemote : Module() {
+    @Setting(comment = "When disabled, anybody can send commands")
+    private var friendsOnly = true
+
+    init {
+        hidden = true
+        BaritoneIntegration {
+            hidden = false
+
+            val listener: Listener<PacketEvent.Send>?
+            listener = Listener({ event ->
+                if (event.packet !is ChatMessageC2SPacket) return@Listener
+                val message = event.packet.chatMessage
+
+                if (MessageDetectionHelper.isDirectReceived(true, message)) {
+                    val username = MessageDetectionHelper.getDirectUsername(message) ?: return@Listener
+                    val command = message.detectAndRemove(Regexes.DIRECT)
+                        ?: message.detectAndRemove(Regexes.DIRECT_ALT_1)
+                        ?: message.detectAndRemove(Regexes.DIRECT_ALT_2) ?: return@Listener
+
+                    val prefix = BaritoneIntegration.prefix ?: return@Listener
+                    if (!command.startsWith(prefix) || !isValidUser(username)) return@Listener
+
+                    mc.player?.sendChatMessage(command)
+                }
+            })
+
+            KamiMod.EVENT_BUS.subscribe(listener)
```

Comment:
wouldnt this always call the listener even when the module is disabled? consider subscribing and unsubscribing the listener in `onEnable` and in `onDisable` respectively

Reply:
Wouldn't an `if isDisabled` be neater?

Reply:
i personally dont think it would be. also that would mean theres still a listener getting called on each packet even tho its doing nothing.

Reply:
please use the fix that @zeroeightysix suggested.